pkg/operator/ceph/controller/spec.go

```diff
@@ -75,21 +75,64 @@ var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
 PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/
+LOG_MAX_SIZE=%s
+
+# size of log after removing any alphabet in the end. eg: for '1G' getting '1'
+LOG_MAX_SIZE_IN_DIGIT=$(echo "${LOG_MAX_SIZE}" | sed 's/.$//')
+# get last char of file size. eg: for '1G' getting 'G'
+GET_LAST_CHAR_OF_LOG_SIZE=$(echo $LOG_MAX_SIZE | rev | cut -c1)
+
+if [ "$GET_LAST_CHAR_OF_LOG_SIZE" == "G" ]; then
+	LOG_MAX_SIZE_IN_KB=$(( (LOG_MAX_SIZE_IN_DIGIT *1024*1024)))
+elif [ "$GET_LAST_CHAR_OF_LOG_SIZE" == "M" ]; then
+	LOG_MAX_SIZE_IN_KB=$(( (LOG_MAX_SIZE_IN_DIGIT *1024)))
+fi
+
+LAST_ROTATION_TIME_IN_SEC=$(echo $SECONDS)
 
 if [ -z "$PERIODICITY" ]; then
 	PERIODICITY=24h
 fi
 
+# remove last char from PERIODICITY eg. for 24h getting 24.
+PERIODICITY_IN_DIGIT=$(echo $PERIODICITY | awk '{print $1}' | sed 's/.$//')
+
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
+	SIZE_OF_FILE=$(du --summarize --human-readable $DAEMON_LOG_FILE | awk '{print $1}')
+
+	# size of file after removing any alphabet in the end. eg: for '1G' getting '1'
+	SIZE_OF_FILE_IN_DIGIT=$(du --summarize --human-readable $DAEMON_LOG_FILE | awk '{print $1}' | sed 's/.$//')
+	# get last char of file size. eg: for '1G' getting 'G'
+	GET_LAST_CHAR_OF_FILE_SIZE=$(echo $SIZE_OF_FILE | rev | cut -c1)
+
+	if [ "$GET_LAST_CHAR_OF_FILE_SIZE" != "G" ] && [ "$GET_LAST_CHAR_OF_FILE_SIZE" != "M" ]; then
+		SIZE_OF_FILE_IN_KB=0
+	elif [ "$GET_LAST_CHAR_OF_FILE_SIZE" == "G" ]; then
+		SIZE_OF_FILE_IN_KB=$(echo $SIZE_OF_FILE_IN_DIGIT *1024 | awk '{print $1*1024}'| cut -d $(locale decimal_point) -f 1)
```

Comment:
using `awk` or math op and not build-in math because built throughs error when the calculation is in decimal and here the size of the file can be in decimal.

Same for L120